python-package/lightgbm/basic.py

```diff
@@ -267,26 +268,25 @@ def _data_from_pandas(data, feature_name, categorical_feature, pandas_categorica
             for col, category in zip_(cat_cols, pandas_categorical):
                 if list(data[col].cat.categories) != list(category):
                     data[col] = data[col].cat.set_categories(category)
-        if len(cat_cols):  # cat_cols is pandas Index object
+        if len(cat_cols):  # cat_cols is list
             data = data.copy()  # not alter origin DataFrame
             data[cat_cols] = data[cat_cols].apply(lambda x: x.cat.codes).replace({-1: np.nan})
         if categorical_feature is not None:
             if feature_name is None:
                 feature_name = list(data.columns)
             if categorical_feature == 'auto':
-                categorical_feature = list(cat_cols)
+                categorical_feature = cat_cols_not_ordered
             else:
-                categorical_feature = list(categorical_feature) + list(cat_cols)
+                categorical_feature = list(categorical_feature) + cat_cols_not_ordered
```

Comment:
I wonder whether this is a good decision to add default categorical columns even in case user explicitly specify categorical features. Firstly, this feature is not documented. Secondly, it seems that this approach makes some discomfort for users: the only way to disable default cat features is to convert them in source dataframe. I think it's more comfortable just not include them in list of cat features. The intuition behind this is that user want **either** use default cat features in cat columns, **or** specify cat features in param.

Reply:
good point, this indeed is not comfortable.

Reply:
@guolinke Cool! I'll address this in next PR, OK?

Reply:
yeah, sure